File: 2017/11-Nov/11-14-Sorting-Lists/sort.php
<?php

// Author: Andrew Jarombek
// Date: 11/14/2017
// Sort an array of objects in PHP with a comparator function

class Person 
{
  public $first;
  public $last;
  
  public function __construct($first, $last) 
  {
    $this->first = $first;
    $this->last = $last;
  }
}

$array = array(new Person('Andrew', 'Jar.'), new Person("Thomas", "Cau."),
                new Person("Joe", "Smi."), new Person("Ben", "Fis."));
  
print_r($array);

// Define a comparison function that will be used to sort an array of People by last name 
function comparator($a, $b) {
  return $a->last < $b->last ? -1 : 1; 
}

// Sort with the comparison function
uasort($array, 'comparator');

print_r($array);

File: 2017/11-Nov/11-14-Sorting-Lists/sort.py
# Author: Andrew Jarombek
# Date: 11/14/2017
# Sort a list in Python with the sort() function and key function

people = [{"first":"Andrew", "last":"Jar."}, 
    {"first":"Thomas", "last":"Cau."},
    {"first":"Joe", "last":"Smi."},
    {"first":"Ben", "last":"Fis."}]

print(people)

# Function that accesses the value of the dictionary we are sorting on
def last(item):
    return item["last"]

# reverse optional argument can be removed
people.sort(key=last, reverse=False)

print(people)